src/data.py

```diff
@@ -1,192 +1,68 @@
-        extras.__dict__.update(Vibration.calculate_all(self, extras=extras))
-        extras.__dict__.update(Tracer.calculate_all(self, extras=extras))
-
-        return extras
-
-    def _add_shared_variables(self, extras: SimpleNamespace):
-        """Add common shared variables to extras namespace."""
-        extras.n_diffusing = sum(
-            [e.name == extras.diffusing_element for e in self.species])
-        extras.n_steps = len(
-            self.trajectory_coords) - extras.equilibration_steps
-
-        extras.total_time = extras.n_steps * self.time_step
-
-        diffusing_idx = np.argwhere([
-            e.name == extras.diffusing_element for e in self.species
-        ]).flatten()
-
-        extras.diff_coords = self.trajectory_coords[
-            extras.equilibration_steps:, diffusing_idx, :]
-
-    @classmethod
-    def from_vasprun(cls,
-                     xml_file: str | Path,
-                     cache: Optional[str | Path] = None):
-        """Load data from vasprun.xml.
-
-        Parameters
-        ----------
-        xml_file : Path
-            Path to vasprun.xml
-        cache : Optional[Path], optional
-            Path to cache data for vasprun.xml
-
-        Returns
-        -------
-        data : Data
-            Dataclass with simulation data
-        """
-
-        if not cache:
-            cache = Path(str(xml_file) + '.cache')
-
-        if Path(cache).exists():
-            try:
-                return cls.from_cache(cache)
-            except Exception as e:
-                print(e)
-                print('Error reading from cache, reading full VaspRun')
-
-        run = vasp.Vasprun(
-            xml_file,
-            parse_dos=False,
-            parse_eigen=False,
-            parse_projected_eigen=False,
-            parse_potcar_file=False,
-        )
-
-        structure = run.structures[0]
-        trajectory = run.get_trajectory()
-        trajectory.to_positions()
-
-        data = {
-            'structure': structure,
-            'trajectory_coords': trajectory.coords,
-            'species': structure.species,
-            'lattice': structure.lattice,
-            # size of the time step (*1e-15 = in femtoseconds)
-            'time_step': run.parameters['POTIM'] * 1e-15,
-            # temperature of the MD simulation
-            'temperature': run.parameters['TEBEG'],
-            'parameters': run.parameters,
-        }
-
-        # first create object to check pydantic before caching
-        ret = cls(**data)
-
-        if cache:
-            ret.to_cache(cache)
-
-        return ret
+from .trajectory import Trajectory
+
+
+def calculate_all(trajectory: Trajectory,
+                  *,
+                  diffusing_element: str,
+                  known_structure: str | None = None,
+                  diffusion_dimensions: int = 3,
+                  z_ion: float = 1.0,
+                  n_parts: int = 10,
+                  dist_collective: float = 4.5):
```

Comment:
Should we move this function to the `calculate` folder ? It feels a bit weird now in an otherwise empty `data.py`

Reply:
Yeah, agree that makes sense. I didn't quite know what to do with this function :sweat_smile: